### ISSUE TITLE
Setup with "@cloudflare/vitest-pool-workers"?

### ISSUE THREAD

**Issue by tsteckenborn** [evanderkoogh/otel-cf-workers#173]
Hey there,

has someone a working setup in conjunction with "@cloudflare/vitest-pool-workers"?

I'm currently running in different issues, such as:

```
Failed to load url @opentelemetry/sdk-trace-base (resolved id: @opentelemetry/sdk-trace-base)
```

and / or:

```
No such module "Users/.../node_modules/.pnpm/@opentelemetry+resources@1.26.0_@opentelemetry+api@1.9.0/node_modules/@opentelemetry/resources/build/esm/detectors/platform/node/os".
  imported from "Users/.../node_modules/.pnpm/@opentelemetry+resources@1.26.0_@opentelemetry+api@1.9.0/node_modules/@opentelemetry/resources/build/esm/detectors/platform/node/HostDetectorSync.js"
  ```
  
  I'd assume for now that it's due to some configurations on my end. If somebody has a working sample together with the vitest pool workers it would save me some time and I'd be grateful :)

**Comment by jahands** [comment]
I believe this issue is related to https://github.com/cloudflare/workers-sdk/issues/6581

**Comment by evanderkoogh** [comment]
Yeah.. that certainly seems like a similar issue indeed. And not sure if there is anything I could do to fix that in this library.

If the issue is only in `@opentelemetry/sdk-trace-base` and not in `@opentelemetry/api`, or if you do not do any manual instrumentation, you should be able to test the handler without the `instrument` calls.

**Comment by jahands** [comment]
After messing with this for way too long, I found a solution that worked for me. I created a package that bundles otel-cf-workers and @opentelemetry/api using esbuild to target `browser`, which seems to treeshake out all the Node-specific things: https://www.npmjs.com/package/@jahands/otel-cf-workers

It feels kind of hacky, but it was either this or stop using otel :(

**Comment by jfsiii** [comment]
According to https://github.com/cloudflare/workers-sdk/issues/6581#issuecomment-2653472683

This should work as of [v0.6.14](https://github.com/cloudflare/workers-sdk/releases/tag/%40cloudflare%2Fvitest-pool-workers%400.6.14)

```ts
export default defineWorkersConfig({
  test: {
    deps: {
      optimizer: {
        ssr: {
          enabled: true,
          include: ["@microlabs/otel-cf-workers"],
        },
      },
    },
    poolOptions: {
      workers: {
        // ...
      },
    },
  },
});
```

**Comment by tsteckenborn** [comment]
Did somebody test? I'm still encountering an issue when using opentelemetry api

**Comment by tsteckenborn** [comment]
As it's been some time did it work successfully for somebody? For me it seems to still struggle with `@opentelemetry/api`.

**Comment by evanderkoogh** [comment]
I have just added the most basic of tests (can I import the library) with vitest-pool-workers, so I am hoping to be adding more elaborate tests (like actually invoking requests and checking if they a) return the correct response and b) emit the right telemetry.

But my focus right now is just to get some basic functionality going with RPC calls.. so it might be a while.

But just so I am aware, what are the issues you are seeing with `@opentelemetry/api`?

**Comment by tsteckenborn** [comment]
With regards to the error message see here https://github.com/cloudflare/workers-sdk/issues/6581#issuecomment-2674934834
